src/App.js

```diff
@@ -24,62 +24,87 @@ import getToken from './lib/tokens';
 
 function App() {
   const [listId, setListId] = useState(null);
-  const [userToken, setUserToken] = useState('');
 
-  function saveToken(e) {
-    e.preventDefault();
+  function createList() {
     const token = getToken();
-    localStorage.setItem('token', token);
-    setUserToken(token);
+
+    return db
+      .collection('lists')
+      .add({ token: token }) //add a list and assign it the user's token
+      .then((results) => {
+        const { id } = results; //get the firestore-generated id from the created list
+        if (id === undefined) throw new Error('Failed to create list.');
+        return id;
+      })
+      .then((newListId) => {
+        // list successfully created in db, so update listId state and save token to localStorage
+        setListId(newListId);
+        localStorage.setItem('token', token);
+        return true;
+      });
```

Comment:
Updating my suggestion to check for token conflicts in the database first (since suggested code is outdated with the new comments):
```suggestion
    // before attempting to add new list, make sure token hasn't been used before in the db  
    return isTokenValid(token)   
      .then((listExists) => {
        if(listExists) { 
          // if isTokenValid returns true the token is taken, so try again
          createList();
        } else {
          // otherwise the token is not already associated with a list, so we can safely use it
          return db
          .collection('lists')
          .add({ token: token }) // add a list and assign it the user's token
          .then((results) => {
            const { id } = results; // get the firestore-generated id from the created list
            if (id === undefined) throw new Error('Failed to create list.');
            return id;
          })
          .then((newListId) => {
            // list successfully created in db, so update listId state and save token to localStorage
            setListId(newListId);
            localStorage.setItem('token', token);
            return true;
          });
        }
      });
```